File: app/src/main/java/com/absinthe/kage/device/KageObservable.java
package com.absinthe.kage.device;

public abstract class KageObservable {

    abstract public void register(IDeviceObserver observer);
    public abstract void unregister(IDeviceObserver observer);
    abstract protected void notifyFindDevice(Device device);
    abstract protected void notifyLostDevice(Device device);
    abstract protected void notifyDeviceConnected(Device device);
    abstract protected void notifyDeviceDisConnect(Device device);
    protected abstract void notifyDeviceConnectFailed(Device device, int errorCode, String errorMessage);

}

File: app/src/main/kotlin/com/absinthe/kage/device/KageObservable.kt
package com.absinthe.kage.device

abstract class KageObservable {
    abstract fun register(observer: IDeviceObserver?)
    abstract fun unregister(observer: IDeviceObserver?)
    protected abstract fun notifyFindDevice(device: Device)
    protected abstract fun notifyLostDevice(device: Device)
    protected abstract fun notifyDeviceConnected(device: Device)
    protected abstract fun notifyDeviceDisconnect(device: Device)
    protected abstract fun notifyDeviceConnectFailed(device: Device, errorCode: Int, errorMessage: String?)
}